CRM/Financeexportformats/Hook/BuildForm/FinancialBatchExport.php

```diff
@@ -15,12 +15,31 @@ public function __construct($form) {
   }
 
   public function run() {
+    $this->processQuickbooksFormatFromExportModalContext();
     $this->addExtraExportFormats();
   }
 
   /**
-   * Adds extra financial batch export formats
-   * to the exisitng core ones.
+   * CiviCRM versions < 5.39.1 has two ways to export batches, one of them
+   * shows the export formats in a separate form, while the other shows them
+   * in a Modal window, this adds Quickbooks format support to the Modal
+   * window, and allow it to be processed from such context.
+   */
+  private function processQuickbooksFormatFromExportModalContext() {
+    $selectedExportFormat = \CRM_Utils_Request::retrieve('export_format', 'String');
+    if (empty($selectedExportFormat)
+    || !in_array($selectedExportFormat, ['IIF', 'CSV', 'QuickbooksCSV'])
+    ) {
+      return;
+    }
```

Comment:
it is good to have new lines in the code where it might make sense to have them, and it seems to me that  having a new line here make things a little bit more readable.